cmd/lock-rpc-client.go

```diff
@@ -16,56 +16,99 @@
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.RLock", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) RLock(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".RLock", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
 
 // Lock calls write lock RPC.
-func (lockRPCClient *LockRPCClient) Lock(args dsync.LockArgs) (reply bool, err error) {
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.Lock", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) Lock(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".Lock", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
 
 // RUnlock calls read unlock RPC.
-func (lockRPCClient *LockRPCClient) RUnlock(args dsync.LockArgs) (reply bool, err error) {
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.RUnlock", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) RUnlock(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".RUnlock", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
 
 // Unlock calls write unlock RPC.
-func (lockRPCClient *LockRPCClient) Unlock(args dsync.LockArgs) (reply bool, err error) {
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.Unlock", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) Unlock(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".Unlock", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
 
 // ForceUnlock calls force unlock RPC.
-func (lockRPCClient *LockRPCClient) ForceUnlock(args dsync.LockArgs) (reply bool, err error) {
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.ForceUnlock", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) ForceUnlock(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".ForceUnlock", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
 
 // Expired calls expired RPC.
-func (lockRPCClient *LockRPCClient) Expired(args dsync.LockArgs) (reply bool, err error) {
-	lockArgs := newLockArgs(args)
-	err = lockRPCClient.AuthRPCClient.Call("Dsync.Expired", &lockArgs, &reply)
+func (lockRPC *LockRPCClient) Expired(args dsync.LockArgs) (reply bool, err error) {
+	err = lockRPC.Call(lockServiceName+".Expired", &LockArgs{LockArgs: args}, &reply)
 	return reply, err
 }
+
+// NewLockRPCClient - returns new lock RPC client.
+func NewLockRPCClient(host *xnet.Host) (*LockRPCClient, error) {
+	scheme := "http"
+	if globalIsSSL {
+		scheme = "https"
+	}
+
+	serviceURL := &xnet.URL{
+		Scheme: scheme,
+		Host:   host.String(),
+		Path:   lockServicePath,
+	}
+
+	var tlsConfig *tls.Config
+	if globalIsSSL {
+		tlsConfig = &tls.Config{
+			ServerName: host.Name,
+			RootCAs:    globalRootCAs,
+		}
+	}
+
+	rpcClient, err := NewRPCClient(
+		RPCClientArgs{
+			NewAuthTokenFunc: newAuthToken,
```

Comment:
I am not able to see the authToken being set in any headers, as a client we should set it as part of Authorization header such that server can validate. In this change, there is no authentication as such.  please correct me if i am wrong.